chapter_explainability/global_model_analysis.md

```diff
@@ -0,0 +1,265 @@
+
+```{.python .input  n=6}
+from d2l import torch as d2l
+import pandas as pd
+import torch
+import torch.nn as nn
+import seaborn as sn
+from sklearn import model_selection
+```
+
+## A Loan Dataset
+
+In this task, we will build a predictive model to classify whether a borrower will be a loan defaulter or not. A loan is a sum of money that individuals, companies, or organizations borrow from banks or other financial institutions. It is a core business in many financial institutions, and it can help the borrowers overcome their financial constraints and financially manage events such as paying tuition or buying houses. Although being profitable and beneficial for both lenders and borrowers, it carries risks, and loan default is one of the significant risks. Loan default is when a borrower breaks his/her original agreement with a creditor or lender by discontinuing payments. Loan default will lead to financial loss, so it is crucial to assess borrowers' credit risk to see if they will repay the loan in due time.
+
+
+Now let's get a loan dataset for loan defaulter prediction. This dataset is a subset of a Kaggle competition: [lending club](https://www.kaggle.com/datasets/wordsforthewise/lending-club). We have deleted unnecessary columns, imputed missing values, removed duplicate entries, and resampled the imbalanced class. The resulting dataset consists of around ten thousand examples. There are $12$ feature columns including *loan_amnt* (loan amount), *term* (loan term), *int_rate* (interest rate), *annual_inc* (annual income).  The homeownership type is represented with four dummy columns: *MORTGAGE*, *OWN*, *RENT*, and *OTHER*. The *loan_paid* column is the target we are going to predict and it can be either fully-paid (non-defaulter) or charge-off (defaulter). Other columns are described in the Table below. Note that [FICO scores](https://www.fico.com/en/products/fico-score) represent how credit-worthy a person is. 
+
+
+| Column                   | Description                                                                  |
+|--------------------------|------------------------------------------------------------------------------|
+| *fico_range_high*          | The upper boundary range the borrower's FICO at loan origination belongs to. |
+| *last_fico_range_high*    | The upper boundary range the borrower's last FICO pulled belongs to.         |
+| *last_pymnt_amnt*        | The last total payment amount received.                                           |
+| *chargeoff_within_12_mths* | The number of charge-offs within $12$ months.                               |
+
+
+We split the dataset and use $70\%$ of the examples for training and the remaining $30\%$ for testing.
+
+```{.python .input  n=9}
+#@save
+class LoanData(d2l.DataModule):
+    def __init__(self, url=d2l.DATA_URL + 'loan.csv', batch_size=64,
```

Comment:
Since URL and LoanData are 1:1 mapping, don't treat url as an argument and hardcode it in this method.

Reply:
Likewise, if feat_col and target_col are also determinant, hardcode them so the LoanData interface (e.g., args in the constructor) is cleaner.